examples/python/extract-logs-write-to-cloud-logging/main.py

```diff
@@ -0,0 +1,203 @@
+            'user_dev_branch_name': r['user.dev_branch_name'],
+        })
+        output[event_id].add(user_data)
+    return output
+
+
+def group_event_by_event_id(data):
+    output = defaultdict(set)
+    for r in data:
+        event_id = r['event.id']
+        user_data = json.dumps({
+            'event_category': r['event.category'],
+            'event_name': r['event.name'],
+            'event_id': r['event.id'],
+            'event_created_time': r['event.created_time'],
+            'event_sudo_user_id': r['event.sudo_user_id'],
+        })
+        output[event_id].add(user_data)
+    return output
+
+
+def group_all(data):
+    user = group_user_by_event_id(data)
+    model_set = group_model_set_by_event_id(data)
+    event_attribute = group_event_attribute_by_event_id(data)
+    permission = group_permission_by_event_id(data)
+    event = group_event_by_event_id(data)
+
+    event_id_set = set()
+
+    for r in data:
+        event_id_set.add(r['event.id'])
+
+    output = {}
+    for id in event_id_set:
+        output[id] = {
+            'event': list(event[id]),
+            'permission_set': list(permission[id]),
+            'event_attribute': list(event_attribute[id]),
+            'user': list(user[id]),
+            'model_set': list(model_set[id]),
+        }
+    return output
+
+
+def parse_event_attribute(event_attribute):
+    output = {}
+    for data in event_attribute:
+        r = json.loads(data)
+        output[r['event_attribute_name']] = r['event_attribute_value']
+    return output
+
+
+def get_status(data):
+    ea = parse_event_attribute(data)
+    if 'status' in ea:
+        return ea['status']
+    return ''
+
+
+def format(aggregated_data):
+    data = aggregated_data
+    output = []
+
+    for id in aggregated_data:
+
+        output.append({
+            'logName': 'looker_system_activity_logs',
+            'timestamp': json.loads(data[id]['event'][0])['event_created_time'],
+            'insertId': id,
+            'resource': {
+                'type': 'looker',
+            },
+            'protoPayload': {
+                '@type': 'looker_system_activity_logs',
+                'authenticationInfo': {
+                    'principalEmail': json.loads(data[id]['user'][0])['user_email']
+                },
+                'serviceName': 'looker.com',
+                'methodName': json.loads(data[id]['event'][0])['event_name'],
+                'details': parse_event_attribute(data[id]['event_attribute']),
+                'status': get_status(data[id]['event_attribute']),
+            }
+        })
+
+    return output
+
+
+def write_log_entry(formatted_data):
+
+    logging_client = logging.Client()
+    logger = logging_client.logger('looker_system_activity_logs')
+
+    for log in formatted_data:
+        logger.log_struct(log)
+
+    print("Wrote logs to {}.".format(logger.name))
+
+
+if __name__ == "__main__":
+    data = get_looker_data()
+    agg_data = group_all(data)
```

Comment:
It seems like we're grouping on event ID but i think each event is unique. I'll confirm with my colleagues.
If so, this script can be significantly simplified to just transform each event row into the log format you like.